estimagic/visualization/comparison_plot.py

```diff
@@ -0,0 +1,440 @@
+"""
+Draw an interactive comparison plot of named result dictionaries.
+
+The plot can plot many results for large numbers of parameters
+against each other.
+
+The plot can answer the following questions:
+
+1. How are the parameters distributed?
+
+2. How large are the differences in parameter estimates between results
+    compared to the uncertainty around the parameter estimates?
+
+3. Are parameters of groups of results clustered?
+
+"""
+import numpy as np
+import pandas as pd
+from bokeh.layouts import gridplot
+from bokeh.models import BoxSelectTool
+from bokeh.models import ColumnDataSource
+from bokeh.models import HoverTool
+from bokeh.models import TapTool
+from bokeh.models import Title
+from bokeh.models.callbacks import CustomJS
+from bokeh.models.widgets import CheckboxGroup
+from bokeh.plotting import figure
+from bokeh.plotting import show
+
+from estimagic.optimization.utilities import index_element_to_string
+
+
+def comparison_plot(
+    res_dict, color_dict=None, height=None, width=600, point_estimate_plot_kwargs=None
+):
+    """Make a comparison plot from a dictionary containing optimization results.
+
+    Args:
+        res_dict (dict): The keys are the names of different models.
+            Each value is a dictinoary with the following keys and values:
+                - result_df (pd.DataFrame):
+                    params DataFrame returned by maximize or minimize.
+                - model_class (str, optional):
+                    name of the model class to which the model belongs.
+                    This determines the color and checkbox entries with
+                    which model classes can be selected and unselected.
+
+        color_dict (dict):
+            mapping from the model class names to colors.
+
+        height (int):
+            height of the plot.
+
+        width (int):
+            width of the plot.
+    """
+    df = _df_with_all_results(res_dict)
+    heights = _determine_plot_heights(df, height)
+    lower, upper, rect_widths = _create_bounds_and_rect_widths(df)
+    _add_plot_specs_to_df(df, rect_widths, lower, upper, color_dict)
```

Comment:
The function should return df so it is more explicit.

Reply:
Changed in 604f9b37a2dc4bc5123ddc29a8fceba432d60334.